envoy/common/resource.h

```diff
@@ -11,16 +11,11 @@
 namespace Envoy {
 
 /**
- * A handle for use by any resource managers.
+ * A counted resource for use by any resource manager.
  */
-class ResourceLimit {
+class Resource {
```

Comment:
I split up `ResourceLimit` into `Resource` and then `ResourceLimit` uses that and adds a max on top.

It might make sense to use an `Envoy::Stats::PrimitiveGauge` instead and use that in the resource as well. not sure